File: Algorithm I/day1/704.scala
object Solution {
    def search(nums: Array[Int], target: Int): Int = {
        var left: Int = 0
        var right: Int = nums.length - 1
        while (right - left >= 0) {
            var m = (left + right) / 2
            if (nums(m) == target) 
                return m
            if (nums(m) < target) left = m + 1
            else right = m - 1
        }
        return -1
    }
}

File: algorithm-i/day1/704.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    int search(vector<int>& nums, int target) {
        int l = 0;
        int r = nums.size() - 1;
        while (r - l >= 0) {
            int m = (l + r)/2;
            if (nums[m] == target)
                return m;
            if (nums[m] < target)
                l = m + 1;
            else
                r = m - 1;
        }
        return -1;
    }
};